src/System.Threading.Tasks.Dataflow/src/Base/DataflowBlock.cs

```diff
@@ -1353,7 +1353,7 @@ private void CleanupAndComplete(ReceiveCoreByLinkingCleanupReason reason)
                         {
                             // Complete with the received exception
                             var target = (ReceiveTarget<T>)state;
-                            try { target.TrySetException(target._receivedException ?? new Exception()); }
+                            try { target.TrySetException(target._receivedException); }
```

Comment:
Is there a reason to throw generic Exception here? Above there is an assertion " Debug.Assert(_receivedException != null, "We need an exception with which to fault the task.");" so _receivedException should never be null. I also try to simulate a throw Exception() here and the exception bubble out to user code. What do you think?

Reply:
Do we know that the assert is in fact correct, that there's no way _receivedException could be null?  Cleaning up the code is nice, but the `?? new Exception()` really isn't hurting anything, and if there's even the remote chance _receivedException could be null here, leaving the fallback in place is necessary.  If you can prove that _receivedException can never be null here, the fix looks good.

Reply:
I get the point, i'll try to understand if _receivedException can never be null, also another idea could be try/catch and rethrow argumentnull with a more clear errore message, so in remote chance of null the user get a more useful informativo, What do you think?

Reply:
@tarekgh help me to understand, i don't found the right code path. In method "OfferMessage" "status" is local and is assigned to Accepted or DecliningPermanently. DecliningPermanently is assigned only inside catch block where _receivedException is set. How do the code flow to  CleanupAndComplete(ReceiveCoreByLinkingCleanupReason.SourceProtocolError); with _receivedException null?You say "ReceiveCoreByLinkingCleanupReason.ErrorDuringCleanup which not used yet", but i test that is used as a fallback for ReceiveCoreByLinkingCleanupReason.SourceProtocolError, what am I missing?Thanks in advance.